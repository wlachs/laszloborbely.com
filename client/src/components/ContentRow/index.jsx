// React imports
import React from 'react';
import PropTypes from 'prop-types';

function ContentRow(props) {
  const { children } = props;
  return (
    <div className="col-12 d-flex flex-row justify-content-center align-items-center my-3">
      {/* Render row content */}
      {children}
    </div>
  );
}

ContentRow.propTypes = {
  children: PropTypes.oneOfType(
    [PropTypes.string, PropTypes.object, PropTypes.element, PropTypes.any],
  ),
};

ContentRow.defaultProps = {
  children: null,
};

export default ContentRow;
