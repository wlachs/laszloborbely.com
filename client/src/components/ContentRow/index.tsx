// React imports
import React, {type ReactNode} from 'react';

function ContentRow(props: {
	children?: ReactNode;
}) {
	const {children} = props;
	return (
		<div className='col-12 d-flex flex-row justify-content-center align-items-center my-3'>
			{/* Render row content */}
			{children}
		</div>
	);
}

export default ContentRow;
