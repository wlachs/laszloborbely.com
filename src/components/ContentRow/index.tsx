import { ReactElement, type ReactNode } from 'react';

interface ContentRowProps {
	center?: boolean;
	children?: ReactNode;
}

export function ContentRow({
	children,
	center,
}: ContentRowProps): ReactElement {
	const centerClasses = [
		'd-flex',
		'flex-row',
		'justify-content-center',
		'align-items-center',
	];
	const classes = [
		'w-100',
		...(center ? centerClasses : []),
		/* Desktop */
		'mt-lg-1',
		/* Phone */
		'mb-2',
	].join(' ');
	return (
		<div className={classes}>
			{/* Render row content */}
			{children}
		</div>
	);
}
