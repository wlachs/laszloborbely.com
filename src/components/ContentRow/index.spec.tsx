import {cleanup, render, screen} from '@testing-library/react';
import {afterEach, describe, expect, it} from 'vitest';
import ContentRow from './index.tsx';

describe('ContentRow component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should always render child elements', () => {
		const child = 'child';
		const {container} = render(<ContentRow>{child}</ContentRow>);
		expect(screen.getByText(child)).toBeDefined();
		expect(container).toMatchSnapshot();
	});

	it('should align left', () => {
		const {container} = render(<ContentRow/>);
		expect(container.children[0].className).not.toContain('justify-content-center');
		expect(container).toMatchSnapshot();
	});

	it('should align center', () => {
		const {container} = render(<ContentRow center/>);
		expect(container.children[0].className).toContain('justify-content-center');
		expect(container).toMatchSnapshot();
	});
});
