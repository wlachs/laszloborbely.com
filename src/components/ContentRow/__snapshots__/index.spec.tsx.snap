// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentRow component > should align center 1`] = `
<div>
  <div
    class="w-100 d-flex flex-row justify-content-center align-items-center mt-lg-1 mb-2"
  />
</div>
`;

exports[`ContentRow component > should align left 1`] = `
<div>
  <div
    class="w-100 mt-lg-1 mb-2"
  />
</div>
`;

exports[`ContentRow component > should always render child elements 1`] = `
<div>
  <div
    class="w-100 mt-lg-1 mb-2"
  >
    child
  </div>
</div>
`;
